cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureInstanceConnector.java

```diff
@@ -78,7 +78,8 @@ public String getConsoleOutput(AuthenticatedContext authenticatedContext, CloudI
             AzureClient azureClient = ac.getParameter(AzureClient.class);
             String resourceGroupName = azureResourceGroupMetadataProvider.getResourceGroupName(ac.getCloudContext(), vm.getCloudInstance());
             if (vm.getStatus() == InstanceStatus.STARTED) {
-                doReboot(completables, vm, statuses, () -> azureClient.rebootVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId()));
+                doReboot(completables, vm, statuses, () -> azureClient.stopVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId())
+                        .andThen(azureClient.startVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId())));
```

Comment:
I am a little confused as to why there is an async stop followed by an async start. Does `andThen()` wait for the async to complete? If so, why even use async (unless it is referenced elsewhere).

Reply:
Thank you very much for the review. I verified that andThen() is a blocking call and wait for the first async completable to finish. Also it will re-throw any error that is thrown by the first completable. The link to the snippet I use to verify that : https://gist.github.com/christmasferret/3c0a185160ef420526bf882090517ca7